MetaMask/metamask-extension
# Network Form Design and switch to full screen enhancements

Fixes: #12089 #11958 

Explanation:  
Filling out the custom network form is painful because the extension popup will collapse and lose state. 
With this new design upon clicking the Add Network buttons from the Network tab in pop up and fullscreen subheader and the custom RPC from the network dropdown, a new browser window will open with the new Network form saving the user from losing the state every time they move away want to reconfirm the details they are entering.

Manual Testing:

- Navigation:

> 
> 1. In the extension popup, navigate to settings >> network tab>> click `Add Network`
> 2. In Fullscreen mode, navigate to settings >> network tab>> click `Add Network` in the subheader
> 3. In the network dropdown, click `custom RPC` in the network dropdown.

- Network form will open in a new browser open.
- After entering the details, upon clicking the save button the network details will be saved and the user will be taken back to the home page, and a green message appears in the right-hand corner of the page indicating which network is added.
- If the user clicks the `cancel` button or `X` at the top right corner, the user will be taken to the Network tab with details of the current tab displayed in the existing Network form.
